capstone/src/main/webapp/script.js

```diff
@@ -6,3 +6,57 @@ var instance = M.Tabs.init(elem, options);
 //Initialize sidenav
 const slide_menu = document.querySelectorAll(".sidenav");
 M.Sidenav.init(slide_menu, {});
+
+//Get Most Recent Study Sets To Display
+async function most_recent_studysets() {
+  const response = await fetch("/study_set");
+  const studySets = await response.json();
+  var configuredSetsToPost = configureCardsForSearchPage(studySets);
+
+  document.getElementById("results-container").innerHTML = configuredSetsToPost;
+}
+
+//Get Search Request From User and display result
+async function studysets_by_search() {
+  const stringToSearchBy = document.getElementById("search").value;
+  const response = await fetch("/study_set?searchbar=" + stringToSearchBy);
+  const searchResponse = await response.json();
+  var configuredSetsToPost = configureCardsForSearchPage(searchResponse);
+
+  document.getElementById("results-container").innerHTML = configuredSetsToPost;
+}
+
+function configureCardsForSearchPage(studySets) {
+  var configuredCardStudySets = "";
+  for (var studySet of studySets) {
+    var title = studySet.title;
+    var subject = studySet.subject;
+    var numOfCards = studySet.study_set_length;
+    var author = studySet.user_author;
+    var university = studySet.university;
+    configuredCardStudySets += configureCard(
```

Comment:
Pass the entire `studySet` object to this method rather than pulling out each field individually.